src/main/java/com/edmunds/rest/databricks/DTO/scim/user/UserDTO.java

```diff
@@ -0,0 +1,233 @@
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Databricks SCIM user.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc7643#section-4.1">https://tools.ietf.org/html/rfc7643#section-4.1</a>
+ */
+public class UserDTO {
+
+  @JsonProperty("schemas")
+  private final String[] schemas = new String[]{"urn:ietf:params:scim:schemas:core:2.0:User"};
+  @JsonProperty("id")
+  private long id;
+  @JsonProperty("userName")
+  private String userName;
+  @JsonProperty("name")
+  private NameDTO name;
+  //can't be actually customized for the moment. setting username will create an entry (work,username,primary)
+  @JsonProperty("emails")
+  private EmailDTO[] emails;
+  @JsonProperty("active")
+  private boolean active = true;
+  @JsonSerialize(using = UserGroupSerializer.class)
+  @JsonDeserialize(using = UserGroupDeSerializer.class)
+  private GroupDTO[] groups = new GroupDTO[0];
+  @JsonProperty("entitlements")
+  private EntitlementsDTO[] entitlements = new EntitlementsDTO[0];
+  @JsonProperty("displayName")
+  private String displayName;
+
+  public UserDTO() {
+  }
+
+  /**
+   * Builds an user from another one.
+   *
+   * @param from object to copy from
+   */
+  public UserDTO(UserDTO from) {
+    this.id = from.id;
+    this.userName = from.userName;
+    if (from.name != null) {
+      this.name = new NameDTO(from.name);
+    }
+    if (from.emails != null) {
+      emails = Arrays.copyOf(from.emails, from.emails.length);
+    }
+
+    this.active = from.active;
+    if (from.groups != null) {
+      groups = Arrays.copyOf(from.groups, from.groups.length);
+    }
+
+    if (from.entitlements != null) {
+      entitlements = Arrays.copyOf(from.entitlements, from.entitlements.length);
+    }
+    this.displayName = from.displayName;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    UserDTO userDTO = (UserDTO) o;
+    return active == userDTO.active
+        && Objects.equals(userName, userDTO.userName)
+        && Objects.equals(name, userDTO.name)
+        && Arrays.equals(emails, userDTO.emails)
+        && Arrays.equals(groups, userDTO.groups)
+        && Arrays.equals(entitlements, userDTO.entitlements);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = Objects.hash(userName, name, active);
+    result = 31 * result + Arrays.hashCode(emails);
+    result = 31 * result + Arrays.hashCode(groups);
+    result = 31 * result + Arrays.hashCode(entitlements);
+    return result;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
I believe Lombok `@ToString` also allows fieldset customization

Reply:
in the end I've used the default one..we might have to re-think when the Databricks will update the documentation.